src/lib/PreviewPerf.ts

```diff
@@ -0,0 +1,62 @@
+interface PerformancePaintTiming {
+    loadTime?: DOMHighResTimeStamp;
+    renderTime?: DOMHighResTimeStamp;
+}
+
+interface PerformanceReport {
+    fcp?: number;
+    lcp?: number;
+}
+
+export default class PreviewPerf {
+    private fcpObserver: PerformanceObserver;
+
+    private lcpObserver: PerformanceObserver;
+
+    private performance: Performance = performance;
```

Comment:
Should this be `window.performance`?

Reply:
Karma chokes when I use it that way, unfortunately. It doesn't really make a difference.